src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -97,6 +183,39 @@ struct ExeSpaceUtils<Kokkos::Cuda> {
   static TeamPolicy get_team_policy_force_team_size (Int ni, Int team_size) {
     return TeamPolicy(ni, team_size);
   }
+
+
+
+  template <typename TeamMember, class Lambda, typename ValueType, bool Serialize =
+ #ifdef EKAT_TEST_STRICT_FP
+        true
+ #else
+        true //This is a hack for now since EKAT_TEST_STICT_FP is not recognized in src/
```

Comment:
I would remove the default for `Serialize`. Ekat is a library, and defaults in libraries can bite you later on. Let's have the downstream customers explicitly pass the serialize argument. If they are lazy, they can always do

```
template<...>
KOKKOS_INLINE_FUNCTION
void parallel_reduce(...) {
#ifdef SOME_BFB_MACRO
   ekat::parallel_reduce<...,true>(...);
#else
   ekat::parallel_reduce<...,false>(...);
#endif
}
```

Reply:
Ah, if you remove the default (which I would support), then you should move the bool flag to the beginning of the template args list, so that one can do

```
ekat::ExecSpaceUtils<ExecSpace>::parallel_reduce<false>(...);
```
and have all other template args deduced.